java/system-test/system-test-drivers/src/main/java/sleeper/systemtest/drivers/query/WebSocketQueryDriver.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2022-2024 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.systemtest.drivers.query;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import sleeper.clients.QueryWebSocketClient;
+import sleeper.configuration.properties.table.TablePropertiesProvider;
+import sleeper.core.record.Record;
+import sleeper.core.record.serialiser.RecordJSONSerDe;
+import sleeper.core.schema.Schema;
+import sleeper.query.model.Query;
+import sleeper.systemtest.dsl.instance.SystemTestInstanceContext;
+import sleeper.systemtest.dsl.query.QueryAllTablesDriver;
+import sleeper.systemtest.dsl.query.QueryAllTablesSendAndWaitDriver;
+import sleeper.systemtest.dsl.query.QuerySendAndWaitDriver;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+public class WebSocketQueryDriver implements QuerySendAndWaitDriver {
+    private static final Logger LOGGER = LoggerFactory.getLogger(WebSocketQueryDriver.class);
+
+    private final QueryWebSocketClient queryWebSocketClient;
+    private Map<String, CompletableFuture<List<String>>> future = new HashMap<>();
```

Comment:
I was confused by this field's name. Can it be clear that it's indexed by query ID please?

Reply:
This field has been removed by refactoring this class to implement QueryDriver